modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/RetainedContent.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.andes.kernel;
+
+import org.wso2.andes.amqp.AMQPUtils;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ * DisruptorCachedContent has access to content cache built by the disruptor
+ */
+public class RetainedContent implements AndesContent {
+    /**
+     * Content length of the message
+     */
+    private final int contentLength;
+
+    /**
+     * Used to store message content in memory
+     */
+    private Map<Integer, AndesMessagePart> messagePartCache;
+    private long messageID;
+
+    public RetainedContent(Map<Integer, AndesMessagePart> messageParts, int contentLength, long messageID) {
+        this.messagePartCache = messageParts;
+        this.contentLength = contentLength;
+        this.messageID = messageID;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int putContent(int offset, ByteBuffer destinationBuffer) throws AndesException {
+        int written = 0;
```

Comment:
apart from inherited docs. Lets put a comment explain what we are doing here in the implementation.